sources/PTYSession.m

```diff
@@ -3897,15 +3897,40 @@ - (void)appendBrokenPipeMessage:(NSString *)unpaddedMessage {
                   alternateSemantics:YES];
         [terminal updateDefaultChar];
         mutableState.currentGrid.defaultChar = terminal.defaultChar;
+        NSString *dividerChar = [iTermAdvancedSettingsModel sessionEndMessageDividerCharacter];
         int width = (mutableState.width - message.length) / 2;
-        if (width > 0) {
-            [mutableState appendNativeImageAtCursorWithName:@"BrokenPipeDivider"
-                                                      width:width];
+        if (dividerChar.length == 0) {
+            // Use BrokenPipeDivider by default for security
+            if (width > 0) {
+                [mutableState appendNativeImageAtCursorWithName:@"BrokenPipeDivider"
+                                                          width:width];
+            }
+        } else {
+            // Use custom character
+            if (width > 0) {
+                NSMutableString *leftDivider = [NSMutableString string];
+                for (int i = 0; i < width; i++) {
+                    [leftDivider appendString:dividerChar];
+                }
+                [mutableState appendStringAtCursor:leftDivider];
+            }
         }
+        message = [iTermAdvancedSettingsModel sessionEndMessageText] ?: message;
         [mutableState appendStringAtCursor:message];
-        if (width > 0) {
-            [mutableState appendNativeImageAtCursorWithName:@"BrokenPipeDivider"
-                                                      width:(mutableState.width - mutableState.cursorX + 1)];
+        if (dividerChar.length == 0) {
+            if (width > 0) {
+                [mutableState appendNativeImageAtCursorWithName:@"BrokenPipeDivider"
+                                                          width:width];
+            }
+        } else {
+            int rightWidth = mutableState.width - mutableState.cursorX + 1;
```

Comment:
When you change line 3923 don't compute rightwidth twice. Also this should bec onst.